### PR TITLE
[WIP][Routing] Fix the annotation loader taking a class constant as a beginning of a class name

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Branch? | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18633 |
| License | MIT |
| Doc PR | - |

Hold on with the merge. I'd like to see if there should be more test cases added first.

### CODE REVIEW

File: src/Symfony/Component/Routing/Loader/AnnotationFileLoader.php [hunk 1]
```diff
@@ -96,6 +96,8 @@ protected function findClass($file)
             $token = $tokens[$i];
 
             if (!isset($token[1])) {
+                $class = false;
```
Comment (mkruk-u2) [inline src/Symfony/Component/Routing/Loader/AnnotationFileLoader.php]: I'm not sure if this really fixes the problem.

Example:

```
public function doBar()
{
    $isAbc = FileFormType::class === ABC;
}
```

I'll find "ABC" as a class name.

Comment (jakzal) [inline src/Symfony/Component/Routing/Loader/AnnotationFileLoader.php]: As I said this is not ready yet.

**Comment by jakzal** [comment]
I'm offline till June. If anyone wants to take over this PR, feel free to do so. Otherwise I'll finish this off when I'm back.

**Comment by nicolas-grekas** [comment]
Continued in #18907
Thank you @jakzal

**Pull request closed without merging**